blockstore/apps/search/fieldsets.py

```diff
@@ -0,0 +1,120 @@
+"""
+FieldSets for search app.
+"""
+
+from elasticsearch_dsl import (
+    Keyword,
+    InnerDoc,
+    Integer,
+    Text,
+)
+
+from rest_framework import serializers
+
+from .core.fieldsets import FieldSet
+from .core.serializers import FieldSetSerializer
+
+
+class EntityFieldSet(FieldSet):
+    """
+    FieldSet for core data of the entity.
+    """
+
+    name = 'entity'
+
+    class Document(InnerDoc):
+
+        type = Keyword()
+        id = Keyword()  # Since ids of different types of entities may conflict, we do not store them in id field.
```

Comment:
What do you mean, we do not store them in `id` field? This looks like we are storing the ID in the `id` field?

Reply:
Sorry, I meant to say "Since ids of different types of entities may conflict, we do not store them in the base document id field since that needs to be unique."